### PR TITLE
HDDS-5512. Intermittent failure in SCM HA testInstallSnapshot

### PR DESCRIPTION
**Pull request by neils-dev**
## What changes were proposed in this pull request?
`TestSCMInstallSnapshotWithHA#testInstallSnapshot` fails intermittently in CI environment.  PR includes small patch to integration test to fix intermittent problem.  Problem occurs when transactions are added to leader SCM that are then applied to a follower SCM that is placed into the active state from inactive.  Test puts the follower SCM into active state from inactive and expects the follower to install transactions up to leader index when it checks the SCM follower `lastAppliedTermIndex`, however the updates are done with a nonblocking background thread.  Checking the SCM index in this manner is prone to error due to the nonblocking install. 

To fix the intermittent issue, a simple wait and retry construct is used to check the follower index periodically until it updates.  A 3s timeout is used to propagate a fatal error. 

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-5512

## How was this patch tested?
Tested though` hadoop-ozone/dev-support/checks/integration.sh` script in CI environment with environment variables `$ITERATIONS=60, $MAVEN_OPTS: -Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false -Dmaven.wagon.http.retryHandler.class=standard -Dmaven.wagon.http.retryHandler.count=3`

`hadoop-ozone/dev-support/checks/integration.sh  -Dtest=TestSCMInstallSnapshotWithHA#testInstallSnapshot`

Run cat target/integration/summary.txt
Iteration 1 exit code: 0
Iteration 2 exit code: 0
Iteration 3 exit code: 0
Iteration 4 exit code: 0
Iteration 5 exit code: 0
Iteration 6 exit code: 0
Iteration 7 exit code: 0
Iteration 8 exit code: 0
Iteration 9 exit code: 0
...
Iteration 60 exit code: 0

### CODE REVIEW

**Review by adoroszlai — COMMENTED** [review]
Thanks @neils-dev for the fix.  It looks good to me.

Thank you for testing it repeatedly via Github Actions.  I find it a bit hard to follow, because: 
 * Multiple fixes are present on the branch which is tested by scheduled workflow (HDDS-5358).
 * The fix relevant for this PR is not the same commit or cherry-pick of this commit.  It has extra log lines which may or may not affect the results (I guess they don't, but still, it's a difference).

I would like to suggest the following process to make it more transparent next time:
 1. Create a branch for the bug (HDDS-5512 here).  Apply the fix, push for full CI.
 2. From that commit start a new branch and change the workflow definition, removing unnecessary checks and adding iterations.  Push for repeated tests.
 3. Once you have enough repetitions, open the PR from the fix-only branch.

I think this setup makes it trivial to check that the same fix proposed in the PR was verified in the repeated runs.

**Comment by neils-dev** [comment]
> Thanks @neils-dev for the fix. It looks good to me.
> 
> Thank you for testing it repeatedly via Github Actions. I find it a bit hard to follow, because:
> 
> * Multiple fixes are present on the branch which is tested by scheduled workflow ([HDDS-5358](https://issues.apache.org/jira/browse/HDDS-5358)).
> * The fix relevant for this PR is not the same commit or cherry-pick of this commit.  It has extra log lines which may or may not affect the results (I guess they don't, but still, it's a difference).
> 
> I would like to suggest the following process to make it more transparent next time:
> 
> 1. Create a branch for the bug ([HDDS-5512](https://issues.apache.org/jira/browse/HDDS-5512) here).  Apply the fix, push for full CI.
> 2. From that commit start a new branch and change the workflow definition, removing unnecessary checks and adding iterations.  Push for repeated tests.
> 3. Once you have enough repetitions, open the PR from the fix-only branch.
> 
> I think this setup makes it trivial to check that the same fix proposed in the PR was verified in the repeated runs.

Thanks @adoroszlai for the setup to test and validate with the CI for flaky tests.  I will try it out.  The git actions for you mentioned  (HDDS-5358) that I used was actually meant for other jira issues I was running and used to validate in the CI environ the integration tests for the issues - but it's good that you saw it.

**Review by bshashikant — APPROVED** [review]

**Comment by adoroszlai** [comment]
Thanks @neils-dev for the fix, @bshashikant for the review.
